textdb/textdb-api/src/main/java/edu/uci/ics/textdb/api/storage/ICatalogManager.java

```diff
@@ -0,0 +1,36 @@
+package edu.uci.ics.textdb.api.storage;
+
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.common.Schema;
+
+/**
+ * ICatalogManager is the interface for TextDB's catalog manager, 
+ *   which handles the meta data information about each table.
+ * 
+ * @author Zuozhi Wang
+ */
+public interface ICatalogManager {
+    
+    public ICatalogManager getCatalogManager();
+    
+    public boolean tableExists(String tableName);
+    
+    // create a new table, tableName must be unique
+    public void createTable(String tableName, String indexDirectory, Schema schema) throws Exception;
+    
+    // drop a table
+    public void deleteTable(String tableName) throws Exception;
+    
+    // insert a tuple to a table, the primaryAttribute field must be unique
+    public void insertTuple(String tableName, ITuple tuple, String primaryAttribute) throws Exception;
```

Comment:
@chenlica This is a topic worth discussing. 
Join operator faces the similar problem that it needs a unique record ID to identify records. 
Currently we let user specify the primary attribute. 

If TextDB generates a record ID on its own, should TextDB hide it from the user or show it to the user?  If TextDB hides it from the user, then where the ID is used internally? If TextDB shows it to the user, then how should TextDB returns the record ID after each insertion? I definitely think that we need this discussion.